go/test/endtoend/recovery/pitr/shardedpitr_test.go

```diff
@@ -0,0 +1,540 @@
+	require.NoError(t, err)
+
+	// delete the original shard
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("DeleteShard", "ks/0")
+	require.NoError(t, err)
+
+	// Restart vtgate process
+	err = clusterInstance.VtgateProcess.TearDown()
+	require.NoError(t, err)
+
+	err = clusterInstance.VtgateProcess.Setup()
+	require.NoError(t, err)
+
+	clusterInstance.WaitForTabletsToHealthyInVtgate()
+}
+
+func startBinlogServer(t *testing.T, masterTablet *cluster.Vttablet) *binLogServer {
+	bs, err := newBinlogServer(hostname, clusterInstance.GetAndReservePort())
+	require.NoError(t, err)
+
+	err = bs.start(mysqlMaster{
+		hostname: binlogHost,
+		port:     masterTablet.MysqlctlProcess.MySQLPort,
+		username: mysqlUserName,
+	})
+	require.NoError(t, err)
+	return bs
+}
+
+func removeTablets(t *testing.T, tablets []*cluster.Vttablet) {
+	var mysqlProcs []*exec.Cmd
+	for _, tablet := range tablets {
+		proc, _ := tablet.MysqlctlProcess.StopProcess()
+		mysqlProcs = append(mysqlProcs, proc)
+	}
+	for _, proc := range mysqlProcs {
+		err := proc.Wait()
+		require.NoError(t, err)
+	}
+	for _, tablet := range tablets {
+		tablet.VttabletProcess.TearDown()
+	}
+}
+
+func initializeCluster(t *testing.T) (int, error) {
+
+	clusterInstance = cluster.NewCluster(cell, hostname)
+
+	// Start topo server
+	err := clusterInstance.StartTopo()
+	if err != nil {
+		return 1, err
+	}
+	// Start keyspace
+	keyspace := &cluster.Keyspace{
+		Name: keyspaceName,
+	}
+	clusterInstance.Keyspaces = append(clusterInstance.Keyspaces, *keyspace)
+
+	shard := &cluster.Shard{
+		Name: shardName,
+	}
+	shard0 := &cluster.Shard{
+		Name: shard0Name,
+	}
+	shard1 := &cluster.Shard{
+		Name: shard1Name,
+	}
+
+	// Defining all the tablets
+	master = clusterInstance.NewVttabletInstance("replica", 0, "")
+	replica = clusterInstance.NewVttabletInstance("replica", 0, "")
+	shard0Master = clusterInstance.NewVttabletInstance("replica", 0, "")
+	shard0Replica = clusterInstance.NewVttabletInstance("replica", 0, "")
+	shard1Master = clusterInstance.NewVttabletInstance("replica", 0, "")
+	shard1Replica = clusterInstance.NewVttabletInstance("replica", 0, "")
+
+	shard.Vttablets = []*cluster.Vttablet{master, replica}
+	shard0.Vttablets = []*cluster.Vttablet{shard0Master, shard0Replica}
+	shard1.Vttablets = []*cluster.Vttablet{shard1Master, shard1Replica}
+
+	clusterInstance.VtTabletExtraArgs = append(clusterInstance.VtTabletExtraArgs, commonTabletArg...)
+	clusterInstance.VtTabletExtraArgs = append(clusterInstance.VtTabletExtraArgs, "-restore_from_backup", "-enable_semi_sync")
+
+	err = clusterInstance.LaunchCluster(keyspace, []cluster.Shard{*shard, *shard0, *shard1})
+	require.NoError(t, err)
+	// Start MySql
+	var mysqlCtlProcessList []*exec.Cmd
+	for _, shard := range clusterInstance.Keyspaces[0].Shards {
+		for _, tablet := range shard.Vttablets {
+			if proc, err := tablet.MysqlctlProcess.StartProcess(); err != nil {
+				t.Fatal(err)
```

Comment:
we should not use `t.Fatal` , `require.NoError(t, err)` should be used, as with the earlier, the processes launched before will remain as it is, it won't get closed.

Reply:
Done